enterprise/coderd/prebuilds/membership_test.go

```diff
@@ -0,0 +1,113 @@
+package prebuilds_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/quartz"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbgen"
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	agplprebuilds "github.com/coder/coder/v2/coderd/prebuilds"
+	"github.com/coder/coder/v2/enterprise/coderd/prebuilds"
+)
+
+// TestReconcileAll verifies that StoreMembershipReconciler correctly updates membership
+// for the prebuilds system user.
+func TestReconcileAll(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	clock := quartz.NewMock(t)
+
+	// Helper to build a minimal Preset row belonging to a given org.
+	newPresetRow := func(orgID uuid.UUID) database.GetTemplatePresetsWithPrebuildsRow {
+		return database.GetTemplatePresetsWithPrebuildsRow{
+			ID:             uuid.New(),
+			OrganizationID: orgID,
+		}
+	}
+
+	tests := []struct {
+		name                  string
+		includePreset         bool
+		preExistingMembership bool
+	}{
+		// The StoreMembershipReconciler acts based on the provided agplprebuilds.GlobalSnapshot.
+		// These test cases must therefore trust any valid snapshot, so the only relevant functional test cases are:
+		{name: "no presets, no memberships", includePreset: false, preExistingMembership: false},
+		{name: "preset, but no membership", includePreset: true, preExistingMembership: false},
+		{name: "preset, but already a member", includePreset: true, preExistingMembership: true},
+		{name: "member, but no presets", includePreset: false, preExistingMembership: true},
+	}
+
+	for _, tc := range tests {
+		tc := tc // capture
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			db, _ := dbtestutil.NewDB(t)
+
+			defaultOrg, err := db.GetDefaultOrganization(ctx)
+			require.NoError(t, err)
+			backgroundOrg := dbgen.Organization(t, db, database.Organization{})
```

Comment:
nit: Is `backgroundOrg` just meant to simulate unrelated existing memberships? If so, maybe worth adding a quick comment to clarify that it's not part of the test scenario, just there to ensure the reconciler doesn’t interfere with other orgs. Additionally, maybe renaming it to `unrelatedOrg`.